### PR TITLE
Add an option for specifying the time interval of showing messages to the tracing visualizer

### PR DESCRIPTION
**Pull request by byeong-gil**
The tracing visualization program `fedsd` takes command line arguments of the start and end time to let users be able 
to specify the target time interval of the visualized trace.

### CODE REVIEW

File: util/tracing/visualization/fedsd.py [hunk 1]
```diff
@@ -100,6 +100,10 @@
                     help='RTI\'s lft trace file.')
 parser.add_argument('-f','--federates', nargs='+',
                     help='List of the federates\' lft trace files.')
+parser.add_argument('-s', '--start', type=str, nargs=2,
+                    help='Elapsed logical time that targets to start visualization. [time_value time_unit]')
```
Comment (hokeun) [inline util/tracing/visualization/fedsd.py:104]: Nit: How about "Start time of visualization in elapsed logical time"?

File: util/tracing/visualization/fedsd.py [hunk 2]
```diff
@@ -100,6 +100,10 @@
...
+parser.add_argument('-e', '--end', type=str, nargs=2,
+                    help='Elapsed logical time that targets to end visualization. [time_value time_unit]')
```
Comment (hokeun) [inline util/tracing/visualization/fedsd.py:106]: Same here "End time of visualization in elapsed logical time"

File: util/tracing/trace_to_csv.c [hunk 3]
```diff
@@ -116,156 +118,161 @@ size_t read_and_write_trace() {
...
-            latest_time = trace[i].physical_time;
-        }
-        if (object_instance >= 0 && summary_stats[NUM_EVENT_TYPES + object_instance] == NULL) {
-            summary_stats[NUM_EVENT_TYPES + object_instance] = (summary_stats_t*)calloc(1, sizeof(summary_stats_t));
-        }
-        if (trigger_instance >= 0 && summary_stats[NUM_EVENT_TYPES + trigger_instance] == NULL) {
-            summary_stats[NUM_EVENT_TYPES + trigger_instance] = (summary_stats_t*)calloc(1, sizeof(summary_stats_t));
-        }
+        if ((trace[i].logical_time - start_time) >= trace_start_time
+        && (trace[i].logical_time - start_time) < trace_end_time) {
```
Comment (hokeun) [inline util/tracing/trace_to_csv.c:122]: Nit: I think the correct indentation here should be like this:
```
        if ((trace[i].logical_time - start_time) >= trace_start_time
            && (trace[i].logical_time - start_time) < trace_end_time) {
```

File: util/tracing/trace_to_csv.c [hunk 4]
```diff
@@ -55,13 +55,15 @@ int table_size;
...
+    printf("\nUsage: trace_to_csv [options] trace_file_root (with .lft extension)\n\n");
     printf("\nOptions: \n\n");
-    printf("  -f, --fast [true | false]\n");
-    printf("   Whether to wait for physical time to match logical time.\n\n");
+    printf("  -s, --start [time_spec] [units]\n");
+    printf("   The target time to begin tracing.\n\n");
+    printf("  -e, --end [time_spec] [units]\n");
+    printf("   The target time to stop tracing.\n\n");
+    // printf("  -f, --fast [true | false]\n");
+    // printf("   Whether to wait for physical time to match logical time.\n\n");
```
Comment (hokeun) [inline util/tracing/trace_to_csv.c:65]: Is there any reason why '-f' option is commented out? If you're going to support this later, maybe mark it as a TODO or FIXME?

**Review by hokeun — COMMENTED** [review]
Looks nice! Just left minor comments.

File: util/tracing/visualization/fedsd.py (hunk 1 above)
Comment (byeong-gil) [reply util/tracing/visualization/fedsd.py]: Thanks for the review! I changed the comment in [14ecd1c](https://github.com/lf-lang/reactor-c/pull/320/commits/14ecd1c2e76936257403806efe3e378f347d189a).

File: util/tracing/visualization/fedsd.py (hunk 2 above)
Comment (byeong-gil) [reply util/tracing/visualization/fedsd.py]: Resolved in [14ecd1c](https://github.com/lf-lang/reactor-c/pull/320/commits/14ecd1c2e76936257403806efe3e378f347d189a).

File: util/tracing/trace_to_csv.c (hunk 3 above)
Comment (byeong-gil) [reply util/tracing/trace_to_csv.c]: Resolved in [14ecd1c](https://github.com/lf-lang/reactor-c/pull/320/commits/14ecd1c2e76936257403806efe3e378f347d189a).

File: util/tracing/trace_to_csv.c (hunk 4 above)
Comment (byeong-gil) [reply util/tracing/trace_to_csv.c]: I just inherited the original code below. However, the comment does not look like relating to the visualization tool. I simply removed this part in [6ce0e10](https://github.com/lf-lang/reactor-c/pull/320/commits/6ce0e103dd6ce77ec3e7e3645525c68cd70f5a36).  https://github.com/lf-lang/reactor-c/blob/1f97e6a9eb5dd7c82fb1c1bd226ed66283d05137/util/tracing/trace_to_csv.c#L58-L64

**Review by hokeun — APPROVED** [review]
LGTM

**Pull request merged**